### PR TITLE
feat: add fallback logic for verify and echo

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

File: core/android/src/main/kotlin/com/walletconnect/android/internal/common/di/FailoverUtils.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    return chain.proceed(request.newBuilder().url(getFallbackEchoUrl(request.url.toString())).build())
+}
+
+internal fun fallbackVerify(request: Request, chain: Interceptor.Chain): Response {
+    VERIFY_URL = FAIL_OVER_VERIFY_URL
+    wasVerifyFailOvered = true
+    return chain.proceed(request.newBuilder().url(getFallbackVerifyUrl(request.url.toString())).build())
+}
+
+internal fun Scope.fallbackRelay(chain: Interceptor.Chain, request: Request): Response {
```
Comment (TalhaAli00) [inline core/android/src/main/kotlin/com/walletconnect/android/internal/common/di/FailoverUtils.kt:52]: can we make the order of the parameters request then chain for consistency?

File: core/android/src/main/kotlin/com/walletconnect/android/internal/common/di/CoreNetworkModule.kt [hunk 2]
```diff
@@ -86,16 +83,21 @@ fun coreAndroidNetworkModule(serverUrl: String, connectionType: ConnectionType,
...
-                if (request.url.host == DEFAULT_RELAY_URL && isFailOverException(e)) {
-                    SERVER_URL = "$FAIL_OVER_RELAY_URL?projectId=${Uri.parse(SERVER_URL).getQueryParameter("projectId")}"
-                    wasFailOvered = true
-                    chain.proceed(request.newBuilder().url(get<String>(named(AndroidCommonDITags.RELAY_URL))).build())
+                if (isFailOverException(e)) {
+                    when (request.url.host) {
+                        DEFAULT_RELAY_URL.host -> fallbackRelay(chain, request)
+                        DEFAULT_ECHO_URL.host -> fallbackEcho(request, chain)
+                        DEFAULT_VERIFY_URL.host -> fallbackVerify(request, chain)
+                        else -> chain.proceed(request.newBuilder().url(request.url).build())
```
Comment (TalhaAli00) [inline core/android/src/main/kotlin/com/walletconnect/android/internal/common/di/CoreNetworkModule.kt:99]: we don't need to build a new a request here right? You could just do 
```suggestion
                        else -> chain.proceed(request)
```

**Review by TalhaAli00 — APPROVED** [review]

**Review by Elyniss — COMMENTED** [review]
LGTM! I will add that for keys server later

**Review by Elyniss — APPROVED** [review]
LGTM! I will add that for keys server later

**Review by kacperoak — APPROVED** [review]

**Pull request merged**